src/backend/distributed/operations/worker_node_manager.c

```diff
@@ -404,6 +405,25 @@ NodeIsPrimaryWorker(WorkerNode *node)
 }
 
 
+/*
+ * CoordinatorAddedAsWorkerNode returns true if coordinator is added to the
+ * pg_dist_node. This function also acquires ShareLock on pg_dist_node
+ * and does not release it to ensure that existency of the coordinator in
+ * metadata won't be changed until the end of transaction.
+ */
+bool
+CoordinatorAddedAsWorkerNode()
+{
+	bool groupContainsNodes = false;
+
+	LockRelationOid(DistNodeRelationId(), ShareLock);
```

Comment:
What was the difference between not releasing a lock and releasing it with `NO_LOCK`?

Reply:
we can't unlock a relation with `NO_LOCK`